### ISSUE TITLE
Duplicate declaration error is thrown on `const` declaration in nested async functions

### ISSUE THREAD

**Issue by JLHwung** [babel/babel#8899]
## Bug Report

**Current Behavior**
babel throws

```
Duplicate declaration "foo" (This is an error on an internal node. Probably an internal error.)
```

**Input Code**
- REPL or Repo link if applicable:

https://babeljs.io/repl#?babili=false&browsers=Chrome%20%3E%3D%2045%2Clast%202%20Firefox%20versions%2Cie%20%3E%3D%209%2CEdge%20%3E%3D%2012%2CiOS%20%3E%3D%209%2CAndroid%20%3E%3D%204%2Clast%202%20ChromeAndroid%20versions&build=&builtIns=false&spec=false&loose=false&code_lz=BQQwzgngdgxgBMAlHAvAPjgbwFBzw8aeJVDHffGAeyjABc4AzKq1OAFgCYBuXfAX0RI-g4XiA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=&prettier=false&targets=&version=7.0.0-beta.34&envVersion=7.0.0-beta.34

Note: you have to manually add `@babel/transform-regenerator` 7.0.0 plugin since the url does not synchronize plugin selections.

```js
(async () => {
    (async () => {
      const foo = 42;
    })()
  })()
  
```

**Expected behavior/code**
Compiled as regenerator calls

**Babel Configuration (.babelrc, package.json, cli command)**

```js
module.exports = {
    plugins: [
        "@babel/transform-regenerator"
    ]
}
```

**Environment**
- Babel version(s): v7.0.0
- Node/npm version: Node 10
- OS: macOS 10.14
- Monorepo no
- How you are using Babel: cli

**Possible Solution**
Workaround: add `await` to the nested async function.

**Additional context/Screenshots**
Possibly related: #8525

**Comment by 3DEsprit** [comment]
I just ran across this as well with a very similar setup (babel 7 cli/Node 10), by using one level of nested async functions which the top-level functions were taking in the results of the lower-level. 
I also found it would also error on **let**, but does not with _var_. 

Babel-cli: 7.1.2
Babel-plugin-transform-runtime: 7.1.0
Node: 10.6.0
OSX: 10.14

Adding await to the function was not a possible resolve in my case, since it was written that way to begin with. The only way I could get around the error was removing the variable expression itself. You can always replace **const** and **let** with _var_, but there are many cases where this may be less ideal.

**Comment by JLHwung** [comment]
This issue possibly duplicates #1617 and might be marked as won't fix.

**Comment by nicolo-ribaudo** [comment]
Definitely not "wontfix". Wether the fix needs to be in Babel or Regenerator, it should be done.

**Comment by JLHwung** [comment]
Confirmed that this issue has been fixed and can not be reproduced on babel 7.4.0 now.
